### DISCUSSION TITLE
Dummy BGP Routes

### DISCUSSION THREAD

**Discussion by adamcirillo** [ipspace/netlab#1695] [Q&A]
is there a way to insert some dummy ipv4/6 routes into BGP on a router and then use eBGP to peer the routes into the lab to make some routes for the default routing table?

**Comment by jbemmel** [comment]
See https://github.com/ipspace/netlab/blob/379026a2127f2512d6c9b1ecc1d4cbcaa91fcb1b/tests/integration/bgp/04-originate.yml#L19 for an example

**Reply by adamcirillo** [reply]
i was more thinking of a way to say generate a like a couple hundred subnets with random addresses

**Reply by sdargoeuves** [reply]
That's a topic I wanted to explore a few weeks ago, but other priorities took over.
Something like this could work: https://www.reddit.com/r/networking/comments/om4lb2/lab_env_getting_the_full_route_table/
I haven't explored further, exaBgp could be an option. I'll try to find the interesting resources I found whilst digging, if that could be interesting.

**Reply by jbemmel** [reply]
> i was more thinking of a way to say generate a like a couple hundred subnets with random addresses

You could write a plugin to generate ```bgp.originate``` statements along these lines

**Comment by ipspace** [comment]
As @jbemmel said, netlab supports only "static" origination of BGP routes, either via extra loopbacks, **bgp.originate** parameter (only for IPv4) or **[bgp.originate](https://netlab.tools/plugins/bgp.originate/)** plugin (IPv4 and IPv6).

To get a couple hundred subnets, it's probably best to investigate a BGP daemon that would insert routes from whatever source (even with a plugin generating the prefixes the routes would have to be configured on a network device which might take a bit of time).  **bgpipe** might be able to do the trick, see https://blog.ipspace.net/2024/08/netlab-bgpipe/ to get started.
